agents/honeywell_HMR2300/honeywell_HMR2300.py

```diff
@@ -0,0 +1,379 @@
+    """
+    if write_enabled:
+        ser.write(f'*00WE\r'.encode())
+        print(f"Write enabled {decode_list(ser.readlines())}")
+    ser.write(f'{command}\r'.encode())
+    reply = decode_list(ser.readlines())
+    print(f"Sent command {command}, reply is {reply}")
+    return reply
+
+
+def get_setup_parameters(ser):
+    """Get the basic magnetometer parameters
+
+    Parameters
+    ----------
+    ser: object
+        serial object for the magnetometer"""
+    ser.write('*00Q\r'.encode())
+    reply = decode_list(ser.readlines())
+
+    return reply
+
+
+# LabJack agent class
+class HoneywellHMR2300Agent:
+    def __init__(self, agent, ip, port, sample_rate, acq_chunk, baudrate=9600):
+        """
+        baudrate: int
+            Either 9600 or 19200. 9600 is the default after power restart,
+            but 19200 is required to run at sample rates 40Hz or 50Hz.
+        port: int
+            Serial device port.
+        ip: str
+            serial device port.
+        sample_rate: int
+            Sample rate in Hz. Must be 10, 20, 30, 40, or 50
+        acq_chunk: int
+            Number of seconds to continuously take data
+        """
+        self.active = True
+        self.agent = agent
+        self.log = agent.log
+        self.lock = TimeoutLock()
+        self.take_data = False
+        self.ip = ip
+        self.port = port
+        self.sample_rate = sample_rate
+        self.baudrate = baudrate
+        self.acq_chunk = acq_chunk
+        # Set connection parameters
+        ser = serial.serial_for_url(url=f"socket://{self.ip}:{self.port}", baudrate=self.baudrate, timeout=3)
+        ser.write(chr(27).encode())
+        # Send stop streaming command in case still streaming
+        setup = get_setup_parameters(ser)
+        self.log.info(f"Setup parameters: {setup}")
+        self.log.info(f"Connected to HMR2300 on ip{ip}, port {port}, with baudrate {baudrate}")
+
+        # Register main feed. Exclude influx due to potentially high scan rate
+        agg_params = {
+            'frame_length': 60,
+            'exclude_influx': True
+        }
+        self.agent.register_feed('mag_field',
+                                 record=True,
+                                 agg_params=agg_params,
+                                 buffer_time=1)
+
+        # Register downsampled feed for influx.
+        agg_params_downsampled = {
+            'frame_length': 60
+        }
+        self.agent.register_feed('mag_field_downsampled',
+                                 record=True,
+                                 agg_params=agg_params_downsampled,
+                                 buffer_time=1)
+
+    def start_acq(self, session, params=None):
+        """
+        Task to start data acquisition.
+
+        Parameters
+        ----------
+        sample_rate: int
+            Sample rate in Hz. Must be 10, 20, 30, 40, or 50
+        acq_chunk: int
+            Number of seconds to continuously take data
+        """
```

Comment:
This docstring (and each other task/process docstring) could use updating to follow the [new guidelines](https://ocs.readthedocs.io/en/develop/developer/agents.html#task-and-process-documentation).